pkg/p2p/libp2p/peer.go

```diff
@@ -77,17 +77,23 @@ func (r *peerRegistry) peers() []p2p.Peer {
 	return peers
```

Comment:
in `peers()`, `r.mu.Lock()` should be changed to `r.mu.RLock()` (or switch to a normal mutex, not `RWMutex`)

Reply:
Ah, I think my autosuggestion prefers `RWMutex` for some reason, I made a same mistake couple of times already 🙈

Reply:
Oh, actually, I thought this was something else. I don't think this is a part of this PR as well.